src/filter/hierarchical.rs

```diff
@@ -423,6 +436,26 @@ impl<Key, Filter, Child> HierarchicalFilters<Key, Filter, Child> {
         self.children.into_iter().flatten().collect()
     }
 
+    pub(crate) async fn for_each<F, R>(&mut self, f: F)
+    where
+        F: Fn(Child) -> R + Copy,
+        R: futures::Future<Output = Child>,
+    {
+        self.children = self
+            .children
+            .drain(..)
```

Comment:
According to documentation `drain` will clear `children` collection. Looks like an error

Reply:
This is done to allow creation of futures from each of the values and processing it in parallel

Reply:
We recreate leaves with the new data

Reply:
This was to allow for ownership transfer into closure. Now function is removed